tools/editliveos

```diff
@@ -2093,33 +2114,142 @@ s/\s+rw\s+|\s+rw$/ /g'''.format(self.fslabel, self.releasefile, iso_boot)
+                else:
+                    raise CreatorError("Failed to find environment '%s' : %s" %
+                                       (env, e))
+
+        for group in kickstart.get_groups(self.ks):
+            if group.name == 'core' or group.name in excludedGroups:
+                continue
+
+            try:
+                dbo.selectGroup(group.name, excludedPkgs, group.include)
+                logging.info("selected group: %s", group.name)
+            except dnf.exceptions.MarkingError as e:
+                if kickstart.ignore_missing(self.ks):
+                    logging.warning("Skipping missing group '%s'" %
+                                    (group.name,))
+                else:
+                    raise CreatorError("Failed to find group '%s' : %s" %
+                                       (group.name, e))
+
+        for pkg_name in set(excludedPkgs):
+            dbo.deselectPackage(pkg_name)
+            logging.info("excluding package: '%s'", pkg_name)
+
+        for pkg_name in set(kickstart.get_packages(self.ks,
+                                                   self._get_required_packages())) - set(excludedPkgs):
+            try:
+                dbo.selectPackage(pkg_name)
+                logging.info("selected package: '%s'", pkg_name)
+            except dnf.exceptions.MarkingError as e:
+                if kickstart.ignore_missing(self.ks):
+                    logging.warning(
+                        "Skipping missing package '%s'" % (pkg_name,))
+                else:
+                    raise CreatorError("Failed to find package '%s' : %s" %
+                                       (pkg_name, e))
 
     def install_rpms(self):
+        """Install packages into the install root.
+
+        This function installs the packages listed in the supplied kickstart
+        into the install root. The packages are installed from the
+        repository URLs specified in the kickstart.
+
+        """
+        dnf_conf = self._mktemp(prefix = "dnf.conf-")
+
+        print("dbo creation")
+        dbo = DnfLiveCD(releasever=self.releasever, useplugins=self.useplugins)
+        print("dbo setup start")
+        dbo.setup(dnf_conf, self._instroot, cacheonly=self.cacheonly,
+                   excludeWeakdeps=self.excludeWeakdeps)
+        dbo.conf.substitutions['basearch'] = args.arch
+        dbo.conf.substitutions['releasever'] = self.releasever
+        dbo.conf.releasever = self.releasever
+        print("dbo setup done")
+
+        for repo in kickstart.get_repos(self.ks):
+            (name, baseurl, mirrorlist, proxy, inc, exc, cost, sslverify) = repo
+
+            yr = dbo.addRepository(name, baseurl, mirrorlist)
+            if inc:
+                yr.includepkgs = inc
+            if exc:
+                yr.exclude = exc
+            if proxy:
+                yr.proxy = proxy
+            if cost is not None:
+                yr.cost = cost
+            yr.sslverify = sslverify
```

Comment:
Wouldn't the `imgcreate.dnfinst` module work here instead of duplicating the code?